### PR TITLE
schedulers add glide noising schedule

### PR DESCRIPTION
**Pull request by williamberman**
While testing schedulers for https://github.com/huggingface/diffusers/pull/2297

Adding the glide noising schedule to schedulers which don't support it yet.

I also added prediction_type "sample" to a few of the schedulers.

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_euler_ancestral_discrete.py [hunk 1]
```diff
@@ -214,6 +247,8 @@ def step(
         elif self.config.prediction_type == "v_prediction":
             # * c_out + input * c_skip
             pred_original_sample = model_output * (-sigma / (sigma**2 + 1) ** 0.5) + (sample / (sigma**2 + 1))
+        elif self.config.prediction_type == "sample":
+            raise NotImplementedError("prediction_type not implemented yet: sample")
```
Comment (williamberman) [inline src/diffusers/schedulers/scheduling_euler_ancestral_discrete.py:251]: On schedulers where I couldn't verify good outputs, I threw an explicit error that implies we should support prediction_type sample in the future. I'm not sure if this is correct or there's something inherent about the schedulers which makes them not support the sample prediction type.

File: src/diffusers/schedulers/scheduling_lms_discrete.py [hunk 2]
```diff
@@ -44,6 +45,35 @@ class LMSDiscreteSchedulerOutput(BaseOutput):
     pred_original_sample: Optional[torch.FloatTensor] = None
 
 
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_lms_discrete.py:48]: Can we add a #Copied from here?

File: src/diffusers/schedulers/scheduling_k_dpm_2_discrete.py [hunk 3]
```diff
@@ -21,6 +22,35 @@
 from .scheduling_utils import KarrasDiffusionSchedulers, SchedulerMixin, SchedulerOutput
 
 
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999) -> torch.Tensor:
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_k_dpm_2_discrete.py:25]: Can we add a #Copied from here?

File: src/diffusers/schedulers/scheduling_k_dpm_2_ancestral_discrete.py [hunk 4]
```diff
@@ -22,6 +23,35 @@
 from .scheduling_utils import KarrasDiffusionSchedulers, SchedulerMixin, SchedulerOutput
 
 
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999) -> torch.Tensor:
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_k_dpm_2_ancestral_discrete.py:26]: Can we add a #Copied from here?

File: src/diffusers/schedulers/scheduling_heun_discrete.py [hunk 5]
```diff
@@ -21,6 +22,35 @@
 from .scheduling_utils import KarrasDiffusionSchedulers, SchedulerMixin, SchedulerOutput
 
 
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999) -> torch.Tensor:
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_heun_discrete.py:25]: Can we add a #Copied from here?

File: src/diffusers/schedulers/scheduling_euler_discrete.py [hunk 6]
```diff
@@ -45,6 +46,35 @@ class EulerDiscreteSchedulerOutput(BaseOutput):
     pred_original_sample: Optional[torch.FloatTensor] = None
 
 
+def betas_for_alpha_bar(num_diffusion_timesteps, max_beta=0.999):
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_euler_discrete.py:49]: Can we add a #Copied from here?

**Review by patrickvonplaten — APPROVED** [review]
Cool looks good, let's just try to add # Copied from statements everywhere

**Pull request merged**